{"comments": {}, "bugs": {"685631": {"comments": [{"author": "jorendorff@mozilla.com", "creation_time": "2011-09-08T19:04:52Z", "time": "2011-09-08T19:04:52Z", "creator": "jorendorff@mozilla.com", "count": 0, "id": 5708631, "is_private": false, "tags": [], "bug_id": 685631, "text": "onExceptionUnwind isn't great, it's just what we've got.\n\nhttps://wiki.mozilla.org/Debugger#Debugging_hooks proposes two possible improvements.\n\nOne is to add a hook, dbg.onThrow(frame, value), which is called only when an exception is thrown. The other is to try to provide a complete picture of what's going on by making the hook dbg.onException(status, frame, value) where status is a string like \"throw\", \"catch\", or \"finally\".", "raw_text": "onExceptionUnwind isn't great, it's just what we've got.\n\nhttps://wiki.mozilla.org/Debugger#Debugging_hooks proposes two possible improvements.\n\nOne is to add a hook, dbg.onThrow(frame, value), which is called only when an exception is thrown. The other is to try to provide a complete picture of what's going on by making the hook dbg.onException(status, frame, value) where status is a string like \"throw\", \"catch\", or \"finally\".", "attachment_id": null}]}}}